py/surveyqa/summary.py

```diff
@@ -140,11 +140,10 @@ def get_median(attribute, exposures):
     Output:
         returns a numpy array of the median values for each night
     '''
-    night = exposures['NIGHT']
     medians = []
-    for n in list(OrderedDict(Counter(night)).keys()):
-        exp_night = exposures[exposures['NIGHT'] == n]
-        attrib = exp_night[attribute]
+    for n in np.unique(exposures['NIGHT']):
+        ii = (exposures['NIGHT'] == n)
+        attrib = np.asarray(exposures[attribute])[ii]
```

Comment:
3 optimizations here:
  1. `np.unique` is a faster route to getting a sorted list of unique items, rather than going via `Counter`.  Also note that `OrderedDict` does *not* sort the inputs, but rather is a dictionary that preserves the order in which items were added (which might not have been sorted)
  2. Apply night filter on only a single column of the table, rather than filtering the entire table and then extracting the column of interest.
  3. Filtering a vanilla numpy array is ~10x faster than filtering a column of an astropy Table.  I was surprised by this, but astropy is infamous for prioritizing syntax beauty and features over performance.